docs/en/concepts-and-designs/mal.md

```diff
@@ -236,14 +236,17 @@ last_server_state_sync_time_in_seconds.tagEqual('production', 'catalog').downsam
 
 ## Metric level function
 
-There are three levels in metric: service, instance and endpoint. They extract level relevant labels from metric labels, then informs the meter-system the level to which this metric belongs.
+They extract level relevant labels from metric labels, then informs the meter-system the level to which this metric belongs.
 
  - `servcie([svc_label1, svc_label2...])` extracts service level labels from the array argument.
  - `instance([svc_label1, svc_label2...], [ins_label1, ins_label2...])` extracts service level labels from the first array argument, 
                                                                         extracts instance level labels from the second array argument.
  - `endpoint([svc_label1, svc_label2...], [ep_label1, ep_label2...])` extracts service level labels from the first array argument, 
                                                                       extracts endpoint level labels from the second array argument.
-
+ - `serviceRelation(DetectPoint, [svc_label1...], [related_svc_label1...])` DetectPoint including `DetectPoint.CLIENT` and `DetectPoint.SERVER`, 
+   extracts `service` labels from the first array argument, extracts `related service` labels from the second array argument.
+   If `DetectPoint.CLIENT` the `related service` is upstream of the `service`.
+   If `DetectPoint.SERVER` the `related service` is downstream of the `service`.
```

Comment:
Should we make the position of `source` and `target` services fixed, i.e. the first array argument being `source service` and the second array argument being `target service`? Because this is a service relation metrics, and the `DetectPoint` should not affect the `source` and `target` services. Also, consider if the `DetectPoint` is a variable (not known constant value), how to determine the position of the 2 labels array arguments?

Reply:
Agree, I think using the consistent order `[source], [target]` makes more sense, and easier to read.

Reply:
`Also, consider if the DetectPoint is a variable (not known constant value), how to determine the position of the 2 labels array arguments?`
Do you mean the `DetectPoint ` might be from metrics labelValue?